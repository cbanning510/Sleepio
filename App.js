import React, {useState} from 'react';

import PickerComponent from './components/PickerComponent';

import durations from './utils';

import {StyleSheet, Text, Alert, TouchableOpacity, View} from 'react-native';

const App = () => {
  const [inBedDuration, setInBedDuration] = useState('');
  const [asleepDuration, setAsleepDuration] = useState('');
  const [calculation, setCalculation] = useState(0);

  const reset = () => {
    setAsleepDuration('');
    setInBedDuration('');
  };

  const calculateAndSend = () => {
    // make mock API POST call:
    if (inBedDuration > 0 && asleepDuration > 0) {
      let result = Math.round(100 * (asleepDuration / inBedDuration));
      try {
        // fetch call here
      } catch (e) {
        console.log(e);
      }
      setCalculation(result);
      reset();
    } else {
      Alert.alert('please select a value > 0');
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Sleepio</Text>
      <Text style={styles.sleepScore}>{`Sleep Score: ${calculation}`}</Text>
      <Text>Duration in bed (hours):</Text>
      <PickerComponent
        items={durations}
        value={inBedDuration}
        setValue={setInBedDuration}
      />
      <Text>Duration asleep (hours):</Text>
      <PickerComponent
        items={durations}
        value={asleepDuration}
        setValue={setAsleepDuration}
      />
      <TouchableOpacity onPress={calculateAndSend} style={styles.button}>
        <Text style={styles.buttonText}>Calculate</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'dodgerblue',
  },
  title: {
    fontSize: 32,
    marginBottom: 40,
  },
  sleepScore: {
    margin: 20,
    fontSize: 20,
  },
  button: {
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
    width: 120,
    padding: 10,
    borderColor: 'white',
    borderWidth: 1,
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
    fontWeight: '600',
  },
});

export default App;
