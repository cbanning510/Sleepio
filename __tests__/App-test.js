import 'react-native';
import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import App from '../App';

it('renders correctly', () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});

it('displays expected UI text', () => {
  const {getAllByText} = render(<App />);
  expect(getAllByText('Sleepio').length).toBe(1);
  expect(getAllByText('Select').length).toBe(2);
  expect(getAllByText('Calculate').length).toBe(1);
  expect(getAllByText('Duration in bed (hours):').length).toBe(1);
  expect(getAllByText('Duration asleep (hours):').length).toBe(1);
});
