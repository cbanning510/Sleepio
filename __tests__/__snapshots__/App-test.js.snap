// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "dodgerblue",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 36,
        "marginBottom": 40,
      }
    }
  >
    Sleepio
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "margin": 20,
      }
    }
  >
    Sleep Score: 0
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "margin": 20,
        "paddingTop": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Duration in bed (hours):
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "margin": 14,
          "opacity": 1,
          "padding": 6,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Select
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "margin": 20,
        "paddingTop": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Duration asleep (hours):
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "margin": 14,
          "opacity": 1,
          "padding": 6,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Select
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Calculate
        </Text>
      </View>
    </View>
  </View>
</View>
`;
